File: rmt/src/test/scala/de/flower/rmt/ui/common/app/TestApplication.java
package de.flower.rmt.ui.common.app;

import de.flower.rmt.ui.app.RMTApplication;
import org.apache.wicket.spring.injection.annot.SpringComponentInjector;
import org.springframework.context.ApplicationContext;

/**
 * @author oblume
 */
public class TestApplication extends RMTApplication {

    private ApplicationContext ctx;

    public TestApplication(ApplicationContext ctx) {
        this.ctx = ctx;
    }

    @Override
    protected SpringComponentInjector getSpringComponentInjector() {
        return new SpringComponentInjector(this, ctx);
    }



}

File: rmt/src/test/scala/de/flower/rmt/ui/app/TestApplication.scala
package de.flower.rmt.ui.app

import org.apache.wicket.spring.injection.annot.SpringComponentInjector
import org.springframework.context.ApplicationContext

/**
 * @author flowerrrr
 */
class TestApplication(val ctx: ApplicationContext) extends RMTApplication {

    protected override def getSpringComponentInjector: SpringComponentInjector = {
        return new SpringComponentInjector(this, ctx)
    }

}